### ISSUE TITLE
gensdf loss not falling

### ISSUE THREAD

**Issue by PaulLiuYZ** [princeton-computational-imaging/Diffusion-SDF#21]
![image](https://github.com/princeton-computational-imaging/Diffusion-SDF/assets/66100217/f876d8e9-fd85-4c75-a7d9-d6403507568c)
![image](https://github.com/princeton-computational-imaging/Diffusion-SDF/assets/66100217/127b9345-6b24-4570-be7a-4f373e0466d3)
![image](https://github.com/princeton-computational-imaging/Diffusion-SDF/assets/66100217/35e00d58-b9a3-4202-8fa8-ede31c454b81)
![image](https://github.com/princeton-computational-imaging/Diffusion-SDF/assets/66100217/ceabc12a-1b04-45b3-9686-4e760e70674a)
![image](https://github.com/princeton-computational-imaging/Diffusion-SDF/assets/66100217/f0a95982-b3ae-424c-b3de-ce21b88c2293)
It seems that the gensdf loss does not decrease, can you give me some advice on the training of the conditional generation of thrid stage?

**Comment by Bathsheba** [comment]
I've seen gensdf fail to progress like this when kld_weight is set too high for the dataset. 🤷‍♀️
Also even if it's working it's slow. How many epochs is this? 
 
Not a dev, just someone else running this code on my own data.

**Comment by gene-chou** [comment]
If the losses don't decrease, it simply means that training has converged. You should visualize the outputs / generations to figure out whether the model has been trained well. If not, then you'd need to tune the hyperparameters and loss weights more extensively. Usually the main issue is with the dataset size. We observed that the more data we had, the easier training was.
